lib/Dialect/Random/IR/RandomOps.td

```diff
@@ -0,0 +1,89 @@
+#ifndef LIB_DIALECT_RANDOM_IR_RANDOMOPS_TD_
+#define LIB_DIALECT_RANDOM_IR_RANDOMOPS_TD_
+
+include "lib/Dialect/Random/IR/RandomDialect.td"
+include "lib/Dialect/Random/IR/RandomTypes.td"
+include "mlir/IR/BuiltinAttributes.td"
+include "mlir/IR/CommonTypeConstraints.td"
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+
+class Random_Op<string mnemonic, list<Trait> traits = [Pure]> :
```

Comment:
I don't think `Pure` is correct for all ops. It might be fine for `random.init_prng` and the distribution ops, but `random.sample` should defintively change every time its called, so can't be `Pure`.